app/src/main/java/io/github/droidkaigi/confsched2018/util/SessionAlarm.kt

```diff
@@ -0,0 +1,70 @@
+package io.github.droidkaigi.confsched2018.util
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.os.Build
+import io.github.droidkaigi.confsched2018.R
+import io.github.droidkaigi.confsched2018.model.Session
+import io.github.droidkaigi.confsched2018.model.toReadableDateTimeString
+import io.github.droidkaigi.confsched2018.presentation.common.broadcastreceiver.NotificationBroadcastReceiver
+import java.util.concurrent.TimeUnit
+
+class SessionAlarm(val context: Context, val session: Session.SpeechSession) {
+    var register: Boolean
+        @Deprecated("This property does not have getter") get() {
+            throw NotImplementedError()
+        }
+        set(value) {
+            if (value) register() else unregister()
+        }
+
+    private fun register() {
```

Comment:
🙋 
I thought it's better to pass `Session.SpeechSession` as method parameter here and instantiate this class by dagger injection. Then you can inject SessionAlarm in ViewModel class.
What do you think?